test/integration-tests/strategies/deltaStrategy.ts

```diff
@@ -299,38 +327,169 @@ describe('Delta Strategy integration test', async () => {
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('3400'), 'sETH');
 
-    //   console.log(sethAfter.sub(sethBefore).toString())
-    // })
+      fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry);
+      expect(fullCloseAmount).to.be.eq(0);
+      await expect(vault.connect(randomUser).reducePosition(positionId, toBN('0.1'))).to.be.revertedWith(
+        'amount exceeds allowed close amount',
+      );
+    });
+
+    it('reduce full position if unsafe position + delta is in range', async () => {
+      // 20% jump
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('3500'), 'sETH');
+      const positionId = await strategy.strikeToPositionId(strikes[3]); // 3400 strike
+      const preReduceBal = await lyraTestSystem.snx.quoteAsset.balanceOf(strategy.address);
+
+      const fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry.sub(10)); //account for time passing
+      expect(fullCloseAmount).to.be.gt(0);
+      await vault.connect(randomUser).reducePosition(positionId, fullCloseAmount);
+      const postReduceBal = await lyraTestSystem.snx.quoteAsset.balanceOf(strategy.address);
+      expect(postReduceBal).to.be.lt(preReduceBal);
+    });
+
+    it('partially reduce position if unsafe position + delta is in range', async () => {
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('3600'), 'sETH');
+      const positionId = await strategy.strikeToPositionId(strikes[3]); // 3400 strike
+      const preReduceBal = await lyraTestSystem.snx.quoteAsset.balanceOf(strategy.address);
+
+      const fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry.sub(10)); //account for time passing
+      expect(fullCloseAmount).to.be.gt(0);
+      await vault.connect(randomUser).reducePosition(positionId, fullCloseAmount.div(2));
+      const postReduceBal = await lyraTestSystem.snx.quoteAsset.balanceOf(strategy.address);
+      expect(postReduceBal).to.be.lt(preReduceBal);
+    });
+
+    it('revert reduce position if unsafe position + close amount too large', async () => {
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('3750'), 'sETH');
+      const positionId = await strategy.strikeToPositionId(strikes[3]); // 3400 strike
+
+      const fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry.sub(10)); //account for time passing
+      expect(fullCloseAmount).to.be.gt(0);
+      await expect(vault.connect(randomUser).reducePosition(positionId, fullCloseAmount.mul(2))).to.be.revertedWith(
+        'amount exceeds allowed close amount',
+      );
+    });
+
+    it('partially reduce position with force close if delta out of range', async () => {
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('4000'), 'sETH');
+      const positionId = await strategy.strikeToPositionId(strikes[3]); // 3400 strike
```

Comment:
I found that the previous test (with 3800) wasn't going through the path of forceClose, so I update the mocked spot price here

Reply:
nice catch, thinking we should eventually add some better tracer functionality (hardhat-tracer doesn't log during reverts and can't use event values during the test)